assignments/lambda-classes.js

```diff
@@ -1 +1,177 @@
+  * `catchPhrase` i.e. `Don't forget the homies`
+* Instructor has the following methods:
+  * `demo` receives a `subject` string as an argument and logs out the phrase 'Today we are learning about {subject}' where subject is the param passed in.
+  * `grade` receives a `student` object and a `subject` string as arguments and logs out '{student.name} receives a perfect score on {subject}'
+*/
+class Instructor extends Person {
+	constructor(data) {
+		super(data);
+		this.specialty = data.specialty;
+		this.favLanguage = data.favLanguage;
+		this.catchPhrase = data.catchPhrase;
+	}
+	demo(subject) {
+		console.log(`Today we are learning about ${subject}`);
+		return `Today we are learning about ${subject}`;
+	}
+	grade(student, subject) {
+		console.log(`${student.name} receives a perfect score on ${subject}`);
+		return `${student.name} receives a perfect score on ${subject}`;
+	}
+	randGrade(student) { // stretch
+		student.grade = Number(student.grade);
+		if(student.grade >= 70) {
+			console.log(`${student.name} just graduated!`);
+			return `${student.name} just graduated!`;
+		}
+		student.grade = Number(student.grade) + Number(((Math.random() * 10) - 5).toFixed(0));
+		console.log(`${student.name} has a ${student.grade}% grade`);
+		return `${student.name} has a ${student.grade}% grade`;
+	}
+}
+let grant = new Instructor({
+	name: 'Grant', location: 'ZA', age: 38,
+	favLanguage: 'JavaScript', specialty: 'Front-end', catchPhrase: `Don't forget the homies`
+});
+let shaz = new Instructor({
+	name: 'Shaz', location: 'US', age: 36,
+	favLanguage: 'CSS', specialty: 'Front-end', catchPhrase: `One time shoe shine`
+});
+let arianna = new Instructor({
+	name: 'Arianna', location: 'UK', age: 29,
+	favLanguage: 'JavaScript', specialty: 'Back-end', catchPhrase: `Master is Sacred`
+});
+/*
+#### Student
+* Now we need some students!
+* Student uses the same attributes that have been set up by Person
+* Student has the following unique props:
+  * `previousBackground` i.e. what the Student used to do before Lambda School
+  * `className` i.e. CS132
+  * `favSubjects`. i.e. an array of the student's favorite subjects ['Html', 'CSS', 'JavaScript']
+* Student has the following methods:
+  * `listsSubjects` a method that logs out all of the student's favoriteSubjects one by one.
+  * `PRAssignment` a method that receives a subject as an argument and logs out that the `student.name has submitted a PR for {subject}`
+  * `sprintChallenge` similar to PRAssignment but logs out `student.name has begun sprint challenge on {subject}`
+*/
+class Student extends Person {
+	constructor(data) {
+		super(data);
+		this.className = data.className;
+		this.favSubjects = data.favSubjects;
+		this.previousBackground = data.previousBackground;
+		
+		this.grade = (Math.random() * 100).toFixed(0); // stretch
+	}
+	listSubjects() {
+		console.log(`${this.name} favourite subjects: ${this.favSubjects}`);
+		return `${this.name} favourite subjects: ${this.favSubjects}`;
```

Comment:
`favSubjects` is an array, to make it a list separated by a comma, you could use, `this.favSubjects.join(',')`